sdk/eventhub/event-hubs/src/eventData.ts

```diff
@@ -377,10 +377,61 @@ export interface EventData {
   };
 }
 
+/**
+ * Asserts that the provided data conforms to the `EventData` interface.
+ *
+ * This function performs runtime checks on the `data` object to ensure it matches the expected
+ * structure and types defined in the `EventData` interface. If any of the checks fail, it throws
+ * an error with a descriptive message indicating the mismatch.
+ *
+ * @param data - The data object to validate as `EventData`.
+ * @throws \{Error\} Throws an error if the data does not conform to the `EventData` interface.
+ */
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export function assertIsEventData(data: any): asserts data is EventData {
+  if (data.contentType !== undefined && typeof data.contentType !== "string") {
```

Comment:
should we allow `null` and normalize to `undefined`? let's check current behavior.

Reply:
The type doesn't allow `null` but it is handled in https://github.com/Azure/azure-sdk-for-js/blob/4fc29fe715d4b78bbc276c364d304a28c067267a/sdk/eventhub/event-hubs/src/eventData.ts#L306-L308

Reply:
Addressed in https://github.com/Azure/azure-sdk-for-js/pull/31330/commits/a421b0a3c6e6a9872358a56a18bce2076c3a4690